### PR TITLE
[ENH]  UI development #56

### LINKED ISSUE #56: UI development and integration
React UI should be developed for nice user experience.

### PR DESCRIPTION
**Pull request by tvilight4**
- Closes #56 

Changes proposed in this pull request:

-The react app for the tool is created with the necessary instructions.
-A droppable for selecting the diagnosis options for the users and updating the json.

## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [ ] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [ ] PR links to GitHub issue with mention `Closes #XXXX`
- [ ] Tests pass
- [ ] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: ui-integration/src/App.js [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+    const blob = new Blob([JSON.stringify(filteredData, null, 2)], { type: "application/json" });
+    const url = window.URL.createObjectURL(blob);
+    const a = document.createElement("a");
+    a.href = url;
+    a.download = "filtered_data.json";
+    document.body.appendChild(a);
+    a.click();
+    a.remove();
+  };
```
Comment (barbarastrasser) [inline ui-integration/src/App.js:169]: I think that could be simplified by using the file saver

```suggestion
const downloadJson = () => {
  if (!filteredData) return;

  const blob = new Blob([JSON.stringify(filteredData, null, 2)], { type: "application/json" });
  saveAs(blob, "filtered_data.json");
};
```

File: ui-integration/src/App.js [hunk 2]
```diff
@@ -0,0 +1,298 @@
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import "bootstrap/dist/css/bootstrap.min.css";
+
```
Comment (barbarastrasser) [inline ui-integration/src/App.js:4]:
```suggestion
import { saveAs } from 'file-saver';
```
for the download json - has to be installed first with `npm install file-saver`

**Review by barbarastrasser — COMMENTED** [review]
The code does not run for me. I can access the user interface and the process is running - so the file is produced correctly, but then when it is done I get 
image

Unfortunately, there are no test at all to check if this is my local setup or some general error. I looked up the documentation if there is anything else to install than what I already have installed but I couldn't find anything. Could you add tests to the ui-integration? Maybe someone else could try out the code as well? - if it works for them the code should be fine.

Additionally, I left some comments for the code - feel free to accept what you like and decline what you don't.

File: Readme.md [hunk 3]
```diff
@@ -187,6 +189,8 @@ If you chose the local deployment or you want to access the container from outsi
 ```
 curl -X POST "http://127.0.0.1:9000/process/?code_system=<snomed | cogatlas>&response_type=<file | json>" 
 -F "file=@<filepath-to-tsv-outside-container>.tsv" 
+ This is the command you want to execute in the interactive terminal session within the container. The input file is the to-be-annotated `.tsv` file and the output file is the `.json` file.
+
```
[the remark this answers is not in the document]
Comment (tvilight4) [reply Readme.md:193]: I didnt work on the Readme this was commited while resolving the merge conflicts

[the remark this answers is not in the document]
Comment (barbarastrasser) [reply Readme.md:193]: Then I think it can just go :)

**Review by barbarastrasser — APPROVED** [review]
I just checked the functionality now and it works as desired - i like the no matching option choice :)

**Pull request merged**